packages/driver/src/cypress/runner.ts

```diff
@@ -1124,21 +1122,6 @@ export default {
     specWindow.addEventListener('error', onSpecError('error'))
     specWindow.addEventListener('unhandledrejection', onSpecError('unhandledrejection'))
 
-    // hold onto the _runnables for faster lookup later
-    let _test: any = null
-    let _tests: any[] = []
-    let _testsById: Record<string, any> = {}
```

Comment:
Getting rid of the `_testsById` index will hurt performance when there are a lot of tests since looking up a test will be O(n) instead of O(1). Unless this greatly reduces memory usage, I don't think removing it is worth it.